app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -255,10 +200,75 @@ def update(self, experiment, data):
         return experiment
 
 
+class NimbusStatusValidationMixin:
+    def validate(self, data):
+        data = super().validate(data)
+
+        raised_status_field = None
+        restrictive_statuses = {
+            "status": NimbusConstants.STATUS_CHANGE_ONLY,
+            "publish_status": NimbusConstants.PUBLISH_STATUS_CHANGE_ONLY,
+        }
+
+        if self.instance:
+            status = self.instance.status
+
+            for status_field, statuses in restrictive_statuses.items():
+                instance_value = getattr(self.instance, status_field)
+                if instance_value in statuses and set(data.keys()) != {str(status_field)}:
+                    raised_status_field = (status_field, instance_value)
+
+            if raised_status_field:
```

Comment:
> There are three errors that I'm a little stumped on

One of the errors is in `test_preview_draft_transition_invokes_kinto_task_1_Preview`, where the serializer is not valid and instead reports "Nimbus Experiment has status 'Preview', but can only be changed when set to 'Draft'."

I think the issue is that we've lost [the `if status in self.ALLOWS_STATUS_CHANGE_ONLY:` conditional](https://github.com/mozilla/experimenter/pull/4887/files#diff-da5192d8fdd27a14107b117c1f4fbfbbc6cbee34c61bc19dd273fc612781f7ebL196) in this refactor.

This "raised_status_field" check rejects what *can't* be updated. But it doesn't allow what can. If nothing is raised, it falls through to the next branch of the conditional, which guards against general updates.